src/main/java/javax/jmdns/impl/ServiceInfoImpl.java

```diff
@@ -1200,7 +1200,7 @@ public Collection<DNSRecord> answers(DNSRecordClass recordClass, boolean unique,
         // [PJYF Dec 6 2011] This is bad hack as I don't know what the spec should really means in this case. i.e. what is the class of our registered services.
         if ((recordClass == DNSRecordClass.CLASS_ANY) || (recordClass == DNSRecordClass.CLASS_IN)) {
             // check if the service subtype equals the subtype in the question
-            if (this.getSubtype().isEmpty() && subType != null && this.getSubtype().equals(subType)) {
+            if ((subType == null && ! this.getSubtype().isEmpty()) || this.getSubtype().equals(subType)){
```

Comment:
Is this "!" really correct here? I thought that the subtype must be empty if no `subType` is given in the question?
```suggestion
            if ((subType == null && !this.getSubtype().isEmpty()) || this.getSubtype().equals(subType)) {
```

Reply:
Yes, but there are cases when it must answer with all the subtypes even if not asking for a subtype. 
It was originally  if (this.getSubtype().length() > 0)

Reply:
Your [original issue](https://github.com/jmdns/jmdns/issues/267) states:
> jmdns always includes the subtype in the answer as well, even if not requested

Reply:
My original issue was about **DNSQuestion** where I needed to support. That's why I added copy of the method initially. 
However, the method answers is called from other places as well. e.g Announcer etc

Reply:
ok, thanks for the explanation. Then I'll only do the syntax clean-up suggested above and we should be good to go.